pkg/volume/csi/csi_plugin.go

```diff
@@ -631,6 +648,15 @@ func (p *csiPlugin) CanAttach(spec *volume.Spec) (bool, error) {
 
 // CanDeviceMount returns true if the spec supports device mount
 func (p *csiPlugin) CanDeviceMount(spec *volume.Spec) (bool, error) {
+	supportsStageUnstage, err := p.getSupportsStageUnstage(spec)
+	if err != nil {
+		return false, err
+	}
+
+	if !supportsStageUnstage {
```

Comment:
Does this mean no global path creation for volume types that do not support stage-unstage now?

Reply:
I guess it breaks reconstruction since vol_data.json never gets written at all, I did not see that we have been relying on MountDevice to do that but it makes sense

Reply:
Yes reconstruction should work fine I think. Is there a problem if - we roll this out in an existing cluster, we could potentially leave global paths uncleaned? (i.e kubelet won't remove files that it placed before).

We do need to think about any other code that could potentially be assuming global paths for all CSI drivers. cc @jingxu97

Reply:
> Is there a problem if - we roll this out in an existing cluster, we could potentially leave global paths uncleaned?

yes, but that could only happen if the upgrader neglects to drain the node first so IMO it's fine